tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(60 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("helm manifest are applied", func() {
+			if skipDeploy == "true" {
+				Skip("Skipping the deployment of the operator and the tests")
+			} else {
+				Eventually(deployOperator).Should(Succeed())
+			}
+
+			It("the operator is running", func() {
+				Eventually(getOperatorState).Should(Equal("Running"))
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio control plane", func() {
+		baseDir := filepath.Dir(filepath.Dir(filepath.Dir(wd)))
+		istioVersions, err := getIstioVersions(filepath.Join(baseDir, "versions.yaml"))
+		if err != nil {
+			Fail(fmt.Sprintf("Error getting istio versions from version.yaml file: %v", err))
+		}
+
+		for _, version := range istioVersions {
+			Context(fmt.Sprintf("for supported version %s", version), func() {
+				When(fmt.Sprintf("the istio control plane is installed with version %s", version), func() {
+					BeforeAll(func() {
+						Eventually(createNamespaceIfNotExists).WithArguments(controlPlaneNamespace).Should(Succeed())
+						Eventually(deployIstioControlPlane).WithArguments(version).Should(Succeed())
+
+						DeferCleanup(func() {
+							Eventually(deleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed())
+							Eventually(namespaceIsDeleted).WithArguments(controlPlaneNamespace).Should(Equal(true))
+						})
+					})
+
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(waitForIstioCondition).WithArguments(command, istioName, "Reconciled").Should(Succeed())
+						Eventually(waitForIstioCondition).WithArguments(command, istioName, "Ready").Should(Succeed())
```

Comment:
Here, you're calling `waitForIstioCondition`, which calls `kubectl wait`, but you're also wrapping it in an `Eventually`. So you're actually waiting twice. 

I'd replace `waitForIstioCondition` with `getIstioCondition` (or better yet `getCondition(kind, name, conditionName)` so it can be used for any CRD).

Reply:
Oh, and the `command` parameter should also be removed. The `getCondition` function should use the correct command by itself. I don't think there will be a need to call both `getCondition("kubectl", ...)` and `getCondition("oc", ...)` in the same test. So, there's no point in having to pass the command as an argument.

Reply:
Done